### PR TITLE
V5 - Minor improvements and documentation fixes

### PR DESCRIPTION
**Pull request by juliangut**
Minor documentation fixes:
* Document empty last line
* Missing visibility
* Docblock variable types

Minor styling:
* Excesively long lines
* Docblock spacing
* Unnecessary docblock returns

Improvement:
* Consistent use of EmitterAwareInterface and EmitterAwareTrait

### CODE REVIEW

File: src/Repositories/ClientRepositoryInterface.php [hunk 1]
```diff
@@ -20,11 +20,11 @@
...
      * @param string      $clientIdentifier The client's identifier
-     * @param string|null $clientSecret     The client's secret
+     * @param string      $clientSecret     The client's secret
      * @param string|null $redirectUri      The client's redirect URI
      * @param string|null $grantType        The grant type used
      *
      * @return \League\OAuth2\Server\Entities\Interfaces\ClientEntityInterface
      */
-    public function getClientEntity($clientIdentifier, $grantType, $clientSecret = null, $redirectUri = null);
+    public function getClientEntity($clientIdentifier, $clientSecret, $redirectUri = null, $grantType = null);
```
Comment (alexbilbie) [inline src/Repositories/ClientRepositoryInterface.php]: Client secret will be null in the case of authorization part of auth code grant

**Comment by alexbilbie** [comment]
@juliangut if you can fix that one issue highlighted above will happily merge,  thanks :+1:

**Comment by juliangut** [comment]
Fixing right now.

Another thing, `grantType` will never be null as long as this is always called from a grant which has an identifier (worst case scenario it is an empty string), the parameter should be made mandatory and moved up the list; whether the repository uses it or not is up to the developer

**Comment by alexbilbie** [comment]
I agree

**Comment by juliangut** [comment]
@alexbilbie have you had the time to have a look at least at this PR?

**Comment by alexbilbie** [comment]
@juliangut sorry for the delay on this; could you make Git happy so I can merge this please?

**Comment by juliangut** [comment]
Done!

**Pull request merged**